src/AppInstallerCommonCore/Public/winget/UserSettings.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "AppInstallerStrings.h"
+#include "winget/settings/Source.h"
+#include "winget/settings/Visual.h"
+#include <json.h>
+
+#include <filesystem>
+#include <vector>
+#include <string>
+
+namespace AppInstaller::Settings
+{
+    // The type of argument.
+    enum class UserSettingsType
+    {
+        // Settings files don't exist. A file is created on the first call to the settings command.
+        Default,
+        // Loaded settings.json
+        Standard,
+        // Loaded settings.json.backup
+        Backup,
+    };
+
+    // Representation of the parsed settings file.
+    struct UserSettings
+    {
+        static UserSettings& Instance()
+        {
+            // If we go multi-threaded secure this.
```

Comment:
>// If we go multi-threaded secure this. [](start = 12, length = 39)

It already is secured by what STL (the person) likes to call magic statics.  Basically auto-thread safe init of function local statics.